### PR TITLE
Fix metrics performance regression

### PR DESCRIPTION
**Pull request by yuzhu**
Use AtomicLong with caution, it can have very bad perf when contention is high. 

Ref: https://www.overops.com/blog/java-8-longadders-the-fastest-way-to-add-numbers-concurrently/

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java [hunk 1]
```diff
@@ -119,11 +115,9 @@ public InodeTreePersistentState(InodeStore inodeStore, InodeLockManager lockMana
     mInodeStore = inodeStore;
     mInodeLockManager = lockManager;
     mTtlBuckets = ttlBucketList;
-    mCreatedFileHistogram = null;
-    mRemovedFileHistogram = null;
-    mFileSizeHistogram = new Histogram(2);
-    mRemovedFileRecorder = new Recorder(2);
-    mCreatedFileRecorder = new Recorder(2);
+    mBucketCounter = new BucketCounter(
+        ImmutableList.<Long>builder().add(1024L, 1024L * 1024L, 1024L * 1024L * 1024L,
```
Comment (ggezer) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java:119]: Can you make this configurable?
Leveraging FormatUtils.parseSpaceSize(), we could allow inputs like: "1KB, 1MB, 10MB, 100MB, 1GB, 10GB, 100GB".

File: core/common/src/test/java/alluxio/util/BucketCounterTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    assertEquals(1, counter.getCounters().get(4L).intValue());
+    assertEquals(1, counter.getCounters().get(8L).intValue());
+    counter.insert(6L);
+    assertEquals(1, counter.getCounters().get(4L).intValue());
+    assertEquals(2, counter.getCounters().get(8L).intValue());
+    counter.remove(6L);
+    assertEquals(1, counter.getCounters().get(4L).intValue());
+    assertEquals(1, counter.getCounters().get(8L).intValue());
+  }
+}
```
Comment (ggezer) [inline core/common/src/test/java/alluxio/util/BucketCounterTest.java:39]: nit: can you add a test for MAX_VALUE?

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 3]
```diff
@@ -4489,17 +4488,11 @@ public static void registerGauges(final UfsManager ufsManager, final InodeTree i
...
           inodeTree::getInodeCount);
       MetricsSystem.registerGaugeIfAbsent(MetricKey.MASTER_FILE_SIZE.getName(),
-          () -> StreamSupport.stream(
-              inodeTree.getFileSizeHistogram().logarithmicBucketValues(1024, 1024).spliterator(),
-              false)
-              .map(x -> new Pair<>(
-                  new Pair<>(x.getDoubleValueIteratedFrom(), x.getDoubleValueIteratedTo()),
-                  x.getCountAddedInThisIterationStep()))
-              .collect(Collectors.toMap(Pair::getFirst, Pair::getSecond)));
+          inodeTree::getFileSizeHistogram);
```
Comment (ggezer) [inline core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:4495]: Could you share an example metric output?

File: core/server/master/src/main/java/alluxio/master/metastore/caching/Cache.java [hunk 4]
```diff
@@ -114,35 +113,32 @@ public Cache(CacheConfiguration conf, String name, MetricKey evictionsKey, Metri
    * @return the value, or empty if the key doesn't exist in the cache or in the backing store
    */
   public Optional<V> get(K key, ReadOption option) {
-    try (Timer.Context ctx = MetricsSystem
-        .timer(MetricKey.MASTER_INODE_CACHE_LOAD_TIMER.getName()).time()) {
```
Comment (ggezer) [inline core/server/master/src/main/java/alluxio/master/metastore/caching/Cache.java:118]: Is this a revert?
Can you remove the metric key if so?

**Review by ggezer — CHANGES_REQUESTED** [review]

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java (hunk 3 above)
Comment (yuzhu) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:4495]:
```
value: {
1024: 4,
1048576: 1,
1073741824: 1,
1099511627776: 0,
9223372036854775807: 0
}```

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 5]
```diff
@@ -4489,17 +4488,11 @@ public static void registerGauges(final UfsManager ufsManager, final InodeTree i
       MetricsSystem.registerGaugeIfAbsent(MetricKey.MASTER_FILES_PINNED.getName(),
           inodeTree::getPinnedSize);
       MetricsSystem.registerGaugeIfAbsent(MetricKey.MASTER_FILES_TO_PERSIST.getName(),
-          () -> inodeTree.getToBePersistedIds().size());
+          inodeTree.getToBePersistedIds()::size);
```
[the remark this answers is not in the document]
Comment (yuzhu) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: removed

File: core/common/src/main/java/alluxio/util/BucketCounter.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    mCounter.put(Long.MAX_VALUE, new LongAdder());
+  }
+
+  private LongAdder getStartInterval(Long value) {
+    for (int i = 0; i < mIntervals.size(); i++) {
+      if (mIntervals.get(i) >= value) {
+        return mCounter.get(mIntervals.get(i));
+      }
+    }
+    return mCounter.get(Long.MAX_VALUE);
```
[the remark this answers is not in the document]
Comment (yuzhu) [reply core/common/src/main/java/alluxio/util/BucketCounter.java:49]: not reachable, but java does not know

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java (hunk 1 above)
Comment (yuzhu) [reply core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java]: done

File: core/common/src/test/java/alluxio/util/BucketCounterTest.java (hunk 2 above)
Comment (yuzhu) [reply core/common/src/test/java/alluxio/util/BucketCounterTest.java:43]: done

File: core/server/master/src/main/java/alluxio/master/metastore/caching/Cache.java (hunk 4 above)
Comment (yuzhu) [reply core/server/master/src/main/java/alluxio/master/metastore/caching/Cache.java:118]: done

**Comment by yuzhu** [comment]
@ggezer  addressed comments, PTAL

**Review by ggezer — APPROVED** [review]

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Comment by yuzhu** [comment]
@HelloHorizon Please cherry-pick to branch-2.6.

**Comment by HelloHorizon** [comment]
alluxio-bot, cherry-pick this to branch-2.6 please

**Comment by HelloHorizon** [comment]
alluxio-bot, cherry-pick this to branch-2.6 please

**Pull request merged**